tools/src/main/java/org/aya/util/StringUtil.java

```diff
@@ -26,18 +25,15 @@ public interface StringUtil {
   /**
    * all line separators are treat as 1 character long
    *
-   * @return a (index of the first character, line) list
+   * @return a sequence of integers denoting the start position of each line
    */
-  static @NotNull ImmutableSeq<IntObjTuple2<String>> indexedLines(@NotNull String str) {
-    var lines = ImmutableSeq.from(str.lines());
-    var results = MutableList.<IntObjTuple2<String>>create();
-
+  static @NotNull ImmutableSeq<Integer> indexedLines(@NotNull String str) {
+    var results = MutableList.<Integer>create();
     var index = 0;
-    for (var line : lines) {
-      results.append(IntObjTuple2.of(index, line));
-      index = index + line.length() + 1;    // 1 for the line separator
+    for (var line: ImmutableSeq.from(str.lines())) {
```

Comment:
The construction of `ImmutableSeq` is used for iteration only, which is quite heavy. I think we can:
* use `str.lines().forEach(...)`, but we need to replace `int index` with `MutableIntValue index`.
* use `str.lines().iterator()` with a `while` loop, which is not very pretty.